util/patch/options.go

```diff
@@ -67,3 +74,12 @@ type WithOwnedConditions struct {
 func (w WithOwnedConditions) ApplyToHelper(in *HelperOptions) {
 	in.OwnedConditions = w.Conditions
 }
+
+// WithSanitizedLoggingPaths enables logging for the Patch but replaces the values at each path in the array with "redacted".
+type WithSanitizedLoggingPaths []string
+
+// ApplyToHelper applies this configuration to the given HelperOptions.
+func (w WithSanitizedLoggingPaths) ApplyToHelper(in *HelperOptions) {
```

Comment:
Do we want to have separate options to enable logging and sanitized logging paths?

Or is the idea to force folks to think about sanitizing when enabling logging?

I think just having one option will become awkward if we want to introduce another log-related option in the future.

Reply:
The point here is to force thinking about it more than anything - I think we should get this right though as it will be part of the API. We can always have another apply / option that enables the logging bool by itself.

Reply:
The point here is to force thinking about it more than anything - I think we should get this right though as it will be part of the API. We can always have another apply / option that enables the logging bool by itself.

Reply:
Let's see where the above discussions will lead